app/Controllers/userController.php

```diff
@@ -516,6 +526,30 @@ public function deleteAction() {
 		Minz_Request::forward($redirect_url, true);
 	}
 
+	public function switchAdminAction() {
+		if (!FreshRSS_Auth::hasAccess('admin')) {
+			Minz_Error::error(403);
+		}
+
+		$username = Minz_Request::param('username');
+		if (!FreshRSS_UserDAO::exists($username)) {
+			Minz_Error::error(404);
+		}
+
+		$userConfig = get_user_configuration($username);
+
+		$userConfig->_param('is_admin', !$userConfig->is_admin);
```

Comment:
I have not checked the UI, but can't you pass a boolean instead of toggling?

Reply:
@Alkarex I am not sure I understand your request!

Reply:
This code seems to be toggling the status, right? (i.e. false it it was true, true if it was false)
However, this means that requests are not idempotent, meaning that if a request is sent again (e.g. due to partial connectivity problem), it might not produce the intended effect.
So I am suggesting to pass the desired state as parameter

Reply:
Ok, but the thing is I've wanted to make a toggle. So you mean that the toggle should occur in the call not in the application of the call. Right?

Reply:
Yes, that would be fine :-)

Reply:
I'll change that.

Reply:
I've made some changes. I am hoping that I understood correctly what you meant